discretisedfield/tests/test_mesh.py

```diff
@@ -1688,3 +1688,183 @@ def test_sel_subregions():
+    assert mesh_fft.region.ndim == 3
+    assert np.array_equal(mesh_fft.n, (20, 20, 9))
+    assert np.allclose(mesh_fft.cell, (1, 1, 1))
+    assert np.allclose(mesh_fft.region.pmin, (-10, -10, -4.5))
+    assert np.allclose(mesh_fft.region.pmax, (10, 10, 4.5))
+
+    # 1D mesh with even number of cells
+    p1 = -10
+    p2 = 10
+    cell = 1
+    mesh = df.Mesh(p1=p1, p2=p2, cell=cell)
+
+    mesh_fft = mesh.fftn()
+    assert mesh_fft.region.ndim == 1
+    assert mesh_fft.region.dims == ("k_x",)
+    assert mesh_fft.region.units == ("(m)^-1",)
+    assert np.array_equal(mesh_fft.n, (20,))
+    assert np.allclose(mesh_fft.cell, 0.05)
+    assert np.allclose(mesh_fft.region.pmin, (-0.525,))
+    assert np.allclose(mesh_fft.region.pmax, (0.475,))
+
+    mesh_fft = mesh.fftn(rfft=True)
+    assert mesh_fft.region.ndim == 1
+    assert mesh_fft.region.dims == ("k_x",)
+    assert mesh_fft.region.units == ("(m)^-1",)
+    assert np.array_equal(mesh_fft.n, (11,))
+    assert np.allclose(mesh_fft.cell, 0.05)
+    assert np.allclose(mesh_fft.region.pmin, (-0.025,))
+    assert np.allclose(mesh_fft.region.pmax, (0.525,))
+
+    mesh_fft = mesh.fftn().ifftn()
+    assert mesh_fft.region.ndim == 1
+    assert np.array_equal(mesh_fft.n, (20,))
+    assert np.allclose(mesh_fft.cell, (1,))
+    assert mesh_fft.region.units == ("m",)
+    assert mesh_fft.region.dims == ("x",)
+    assert np.allclose(mesh_fft.region.pmin, p1)
+    assert np.allclose(mesh_fft.region.pmax, p2)
+
+    mesh_fft = mesh.fftn(rfft=True).ifftn(rfft=True)
+    assert mesh_fft.region.ndim == 1
+    assert np.array_equal(mesh_fft.n, (20,))
+    assert np.allclose(mesh_fft.cell, (1,))
+    assert mesh_fft.region.units == ("m",)
+    assert mesh_fft.region.dims == ("x",)
+    assert np.allclose(mesh_fft.region.pmin, p1)
+    assert np.allclose(mesh_fft.region.pmax, p2)
+
+    mesh_fft = mesh.fftn(rfft=True).ifftn(rfft=True, shape=mesh.n)
+    assert mesh_fft.region.ndim == 1
+    assert np.array_equal(mesh_fft.n, (20,))
+    assert np.allclose(mesh_fft.cell, (1,))
+    assert mesh_fft.region.units == ("m",)
+    assert mesh_fft.region.dims == ("x",)
+    assert np.allclose(mesh_fft.region.pmin, p1)
+    assert np.allclose(mesh_fft.region.pmax, p2)
+
+    # 1D mesh with odd number of cells
+    p1 = -10
+    p2 = 9
+    cell = 1
+    mesh = df.Mesh(p1=p1, p2=p2, cell=cell)
+
+    mesh_fft = mesh.fftn(rfft=True).ifftn(rfft=True, shape=mesh.n)
+    assert mesh_fft.region.ndim == 1
+    assert np.array_equal(mesh_fft.n, (19,))
+    assert np.allclose(mesh_fft.cell, (1,))
+    assert mesh_fft.region.units == ("m",)
+    assert mesh_fft.region.dims == ("x",)
+    assert np.allclose(mesh_fft.region.pmin, (-9.5,))
+    assert np.allclose(mesh_fft.region.pmax, (9.5,))
+
+
+@pytest.mark.parametrize(
+    "shape, error",
+    [
+        [(20, 20, 10), ValueError],
+        [(20, 20, 7), ValueError],
+        [(20, 20, 6), ValueError],
```

Comment:
Add brief comments why they fail to the end of the individual lines.
Why do we have the test with `6` and `7`? Do they test different things?